### PR TITLE
add check for own SSH version in /usr/local/bin/

### PR DESCRIPTION
**Pull request by kernelpaniclite**
I added a check for a custom SSH version at /usr/local/bin/

No integration in the GUI but in case a custom SSH version exists, the user probably want to use it.
See also this request: https://github.com/sequelpro/sequelpro/issues/2083

Maybe I or somebody else will iterate on this.

### CODE REVIEW

**Comment by dmoagx** [comment]
I think this is a bad idea:
1. We should not assume that there is an ssh in /usr/local/bin/ just because the user put it there.
Right now this is how my `/user/local/bin` looks:
```
$ ls -1
azure*
azure-uninstall*
capinfos@
dftest@
dumpcap@
editcap@
ksdiff*
mate*
mergecap@
ncat*
ndiff*
nmap*
nmap-update*
nping*
pstorm*
randpkt@
rawshark@
text2pcap@
tshark@
wireshark*
```
The only two things I every gave *explicit* permission to put themselves there are `mate` and `ksdiff`. 
Likewise there could be other apps that put their custom SSH client there because they need one and don't care about software quality ;)

2. It is a security risk. We use a hardcoded path to SSH because otherwise the user might lose control over their password. This would mostly apply to multi-user and networked systems where it is thinkable that someone might replace the ssh in /usr/local to sniff passwords.

3. Compatibility. Sequel Pro only supports the SSH versions that come with OS X. Everything else would probably cause a lot more support efforts – which our small team can't handle.
Thus even if a user willingly installed a custom version of SSH, other applications should still work like before. I'm imagining something like: A user installs a custom ssh, forgets about it, a few weeks later starts Sequel Pro and his connections no longer work. He blames Sequel Pro, because obviously he didn't change a thing™.

To summarize:
The only way we could implement this is via a (possibly hidden) preference. Then again, if we put it in the GUI preferences it would look like we officially support custom SSH builds (which we don't). On the other hand, a hidden preference has (nearly) the same potential risk of it being changed underhand without the user ever noticing as discussed above.

**Comment by kernelpaniclite** [comment]
I understand your position and do also think that this commit can't be a such in a official release.
And I welcome your approach regarding support and safety of the typical user. I intended it so be more of a starting point.

But I also think that a malicious program trying to get on your SSH-keys or passwords could do that also with the system delivered SSH version. Users enter their admin password all the time without know what happens.

I think that a special section in the GUI would be best solution and I would be happy to help. This section could be labeled with unsupported or something like that but would help users with special needs. In this case to increase security needs.

**Comment by dmoagx** [comment]
I've added a GUI option in 9f2e5a6b0f which should provide similar functionality to your patch.
So I'll go ahead and close it now.

**Comment by kernelpaniclite** [comment]
Thanks for implementing and having an open hear:-)
